File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.cpp
class Solution {
public:
   bool checkDistances(string s, vector<int>& dist) {
    for (int i = 0; i < s.size(); ++i) {
        int d = dist[s[i] - 'a'];
        if (i + d + 1 >= s.size() || s[i + d + 1] != s[i])
            return false;
        dist[s[i] - 'a'] = -1;
    }
    return true;
}
};

File: 2399-check-distances-between-same-letters/2399-check-distances-between-same-letters.java
class Solution {
    public boolean checkDistances(String s, int[] dist) {
    for (int i = 0; i < s.length(); ++i) {
        int d = dist[s.charAt(i) - 'a'];
        if (i + d + 1 >= s.length() || s.charAt(i + d + 1) != s.charAt(i))
            return false;
        dist[s.charAt(i) - 'a'] = -1;
    }
    return true;        
}
}